crates/crabe_math/src/shape/line.rs

```diff
@@ -12,3 +12,148 @@ pub struct Line {
     /// The ending point of the line segment.
     pub end: Point2<f64>,
 }
+
+impl Line {
+    pub fn new(start: Point2<f64>, end: Point2<f64>) -> Self {
+        Self { start, end }
+    }
+
+    // return the intersection point between two lines
```

Comment:
Use doc comments instead of comments `\\\` 
And look at another function that comments, the standard documentation is to have 
/// Goal of the function
/// Arguments
/// Returns
/// Some exemple